### PR TITLE
add spec for Dor::Etd::IdentityMetadataGenerator

### LINKED ISSUE #589: need tests for ETD identityMetadataGenerator
https://github.com/sul-dlss/common-accessioning/pull/585/files shows a new class split out for Dor::Etd::IdentityMetadataGenerator.  

It needs tests.  (It probably didn't have tests before it was split out - but that's no excuse).

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made?

Fixes #589 

We had no test for this class, which was created in PR #585

## Was the usage documentation (e.g. README, DevOpsDocs, wiki, queue specific README) updated?

na

### CODE REVIEW

File: lib/dor/etd/identity_metadata_generator.rb [hunk 1]
```diff
@@ -53,17 +57,16 @@ def generate
               unless old_identity_doc.nil?
                 catkey = old_identity_doc.at_xpath('//catkey')
                 catkey = old_identity_doc.at_xpath("//otherId[@name='catkey']") if catkey.nil?
-                xml.text(catkey.text)
+                xml.text(catkey.text) if catkey
```
Comment (ndushay) [inline lib/dor/etd/identity_metadata_generator.rb]: look!  this was a bug I squashed after my tests found it!

Comment (jcoyne) [reply lib/dor/etd/identity_metadata_generator.rb]: The  ETDs are always going to have a catkey, so maybe we don't want a conditional.

Comment (ndushay) [reply lib/dor/etd/identity_metadata_generator.rb]: I wonder if there are any extant ETDs that aren't past this step that either don't have a catkey, or have an empty one.   If both of these are errors, there might be a better way to surface such a problem beyond "no method "text" on nil object".

I defer to your judgement -- it's just that in general, our metadata is in all sorts of states as requirements have changed over time.  For ETDs, I imagine there aren't a lot of old ones stuck in the middle of their accessioning.

We do have SOME, tho.  Perhaps @andrewjbtw can weigh in.

image

**Comment by ndushay** [comment]
@andrewjbtw - do we ever have
-  ETDs with existing IdentityMetadata 
  - that are going through the etd-submit robots 
    - where the existing identityMetadata has an empty catkey?

("no" would be a delightful answer)

**Comment by andrewjbtw** [comment]
I'd love to be able to say "no" but instead the accurate answer is "I have no idea" at this point.

**Comment by ndushay** [comment]
@jcoyne took out the added if clause, added a comment, adjusted specs as necessary.

**Pull request merged**